### PR TITLE
Split PR and add Maisi find mask script

### PR DESCRIPTION
**Pull request by Can-Zhao**
Fixes # .

### Description
Split PR and add Maisi find mask script

### Checks

- [x] Avoid including large-size files in the PR.
- [x] Clean up long text outputs from code cells in the notebook.
- [x] For security purposes, please check the contents and remove any sensitive info such as user names and private key.
- [ ] Ensure (1) hyperlinks and markdown anchors are working (2) use relative paths for tutorial repo files (3) put figure and graphs in the `./figure` folder
- [ ] Notebook runs automatically `./runner.sh -t <path to .ipynb file>`

### CODE REVIEW

File: generative/maisi/scripts/find_masks.py [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+    spacing: Sequence[float] | float = 1.0,
+    output_size: Sequence[int] = [512, 512, 512],
+    check_spacing_and_output_size: bool = False,
+    database_filepath: str = "./data/database.json",
+    mask_foldername: str = "./data/masks/",
+):
+    """
+    Find candidate masks that fullfills all the requirements.
+    They shoud contain all the body region in `body_region`, all the anatomies in `anatomy_list`.
+    If there is no tumor asked in `anatomy_list`, we also expect the candidate masks to be tumor free.
```
Comment (KumoLiu) [inline generative/maisi/scripts/find_masks.py:67]:
```suggestion
    If there is no tumor specified in `anatomy_list`, we also expect the candidate masks to be tumor free.
```

File: generative/maisi/scripts/find_masks.py [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+    If check_spacing_and_output_size is True, the candidate masks need to have the expected `spacing` and `output_size`.
+    Args:
+        body_region: list of input body region string. If single str, will be converted to list of str.
+                     The found candidate mask will include these body regions.
```
Comment (KumoLiu) [inline generative/maisi/scripts/find_masks.py:71]:
```suggestion
        body_region: list of input body region string. If single str, will be converted to list of str.
            The found candidate mask will include these body regions.
```

File: generative/maisi/scripts/find_masks.py [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+    if not os.path.exists(mask_foldername):
+        zip_file_path = mask_foldername + ".zip"
+
+        if not os.path.isfile(zip_file_path):
+            raise ValueError(f"Please download {zip_file_path} following the instruction in ./data/README.md.")
+
+        with zipfile.ZipFile(zip_file_path, "r") as zip_ref:
+            print(f"Extracting {zip_file_path}...")
+            zip_ref.extractall(path=mask_foldername)
+        print(f"Unzipped {zip_file_path} to {mask_foldername}.")
```
Comment (KumoLiu) [inline generative/maisi/scripts/find_masks.py:98]: Looks we can use `extractall` here. 
https://github.com/Project-MONAI/MONAI/blob/58106a6255f5936d26e479d11d3a34e3d536c112/monai/apps/utils.py#L245

BTW, if `zip_file_path` is a file which can be directly download, you can also consider using:
https://github.com/Project-MONAI/MONAI/blob/58106a6255f5936d26e479d11d3a34e3d536c112/monai/apps/utils.py#L304

File: generative/maisi/scripts/find_masks.py [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+            continue
+
+        # extract region indice (top_index and bottom_index) for candidate mask
+        top_index = [index for index, element in enumerate(_item["top_region_index"]) if element != 0]
+        top_index = top_index[0]
+        bottom_index = [index for index, element in enumerate(_item["bottom_region_index"]) if element != 0]
+        bottom_index = bottom_index[0]
+
+        # whether to keep this mask, default to be True.
+        keep_flag = True
```
Comment (KumoLiu) [inline generative/maisi/scripts/find_masks.py:118]: `keep_mask` feels a little more intuitive, but feel free if you don't want to update it.

**Review by KumoLiu — COMMENTED** [review]
Thanks for the PR.

File: generative/maisi/scripts/find_masks.py (hunk 3 above)
Comment (Can-Zhao) [reply generative/maisi/scripts/find_masks.py:98]: Thank you! Done. We may prefer to ask user to download it manually by run `python -m monai.bundle download_large_files --bundle_path .` . What do you think?

Comment (KumoLiu) [reply generative/maisi/scripts/find_masks.py]: Sure, that also works. Thanks.

**Review by KumoLiu — APPROVED** [review]
Thanks for the quick update, looks good to me.

If you don't have any other comments, I will merge this one. cc @Nic-Ma @mingxin-zheng

**Pull request merged**